### PR TITLE
Migrate Form to TypeScript.

### PR DESCRIPTION
**Pull request by huwshimi**
## Done

- Migrate Form to TypeScript.

## Fixes

Fixes: #217.

### CODE REVIEW

File: src/components/Form/Form.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  inline: boolean;
+  stacked?: false;
+};
+
+export type StackedForm = BaseForm & {
+  stacked: boolean;
+  inline?: false;
+};
+
+export type Props = (OrderedForm | InlineForm | StackedForm) &
```
Comment (Caleb-Ellis) [inline src/components/Form/Form.tsx:26]: This seems a bit hard to parse. I wonder if it would be better to update the props if there are actually three mutually exclusive variants, e.g. `appearance: "default" | "inline" | "stacked" and do the same deprecation thing as with notifications.

Comment (huwshimi) [reply src/components/Form/Form.tsx:26]: Yeah I reckon you're right. It's the same thing (only worse) with the list component: https://github.com/canonical-web-and-design/react-components/pull/522.

I suspect as you say we should split these up into individual components, but I'm inclined to want to get these landed as TypeScript files first so that we know if we're breaking things when we split them up (and what the different types should be when we do so).

Comment (Caleb-Ellis) [reply src/components/Form/Form.tsx:26]: That's fair!

**Review by Caleb-Ellis — APPROVED** [review]
